src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -490,8 +490,10 @@ public Void caseStringType() {
         public Void caseObjectType(ObjectType type) {
           if (type.isRecordType()) {
             visitRecordType((RecordType) type);
-          } else {
+          } else if (type.getReferenceName() != null) {
             emit(type.getReferenceName());
+          } else {
+            emit("Object");
```

Comment:
We are inside of caseObjectType here, so I don't think it's necessary to widen to `any`. Maybe there's a test case I'm missing though